src/lib/voiceSearchHelper/__tests__/index-test.ts

```diff
@@ -39,38 +29,54 @@ describe('VoiceSearchHelper', () => {
   });
 
   it('is not supported', () => {
-    const voiceSearchHelper = getVoiceSearchHelper();
+    const voiceSearchHelper = createVoiceSearchHelper({
+      searchAsYouSpeak: false,
+      onQueryChange: () => {},
+      onStateChange: () => {},
+    });
     expect(voiceSearchHelper.isBrowserSupported()).toBe(false);
   });
 
   it('is not listening', () => {
-    const voiceSearchHelper = getVoiceSearchHelper();
+    const voiceSearchHelper = createVoiceSearchHelper({
+      searchAsYouSpeak: false,
+      onQueryChange: () => {},
+      onStateChange: () => {},
+    });
     expect(voiceSearchHelper.isListening()).toBe(false);
   });
 
   it('is supported with webkitSpeechRecognition', () => {
     window.webkitSpeechRecognition = () => {};
-    const voiceSearchHelper = getVoiceSearchHelper();
+    const voiceSearchHelper = createVoiceSearchHelper({
+      searchAsYouSpeak: false,
+      onQueryChange: () => {},
+      onStateChange: () => {},
+    });
     expect(voiceSearchHelper.isBrowserSupported()).toBe(true);
   });
 
   it('is supported with SpeechRecognition', () => {
     window.SpeechRecognition = () => {};
-    const voiceSearchHelper = getVoiceSearchHelper();
+    const voiceSearchHelper = createVoiceSearchHelper({
+      searchAsYouSpeak: false,
+      onQueryChange: () => {},
+      onStateChange: () => {},
+    });
     expect(voiceSearchHelper.isBrowserSupported()).toBe(true);
   });
 
   it('works with mock SpeechRecognition (searchAsYouSpeak:false)', () => {
-    let recognition;
+    const listeners: any = {};
     window.SpeechRecognition = jest.fn().mockImplementation(() => ({
-      start() {
-        /* eslint-disable-next-line consistent-this */
-        recognition = this;
+      start() {},
```

Comment:
We re-use the exact same implementation multiple times inside the test. We could have a function that create the mock version of `SpeechRecognition` (like we do for [the client or the helper](https://github.com/algolia/instantsearch.js/blob/61ae52c9b1b29f812844c00d1b44eca3be929a9f/src/connectors/query-rules/__tests__/connectQueryRules-test.ts#L31-L43)). You can store the [`listeners`](https://github.com/algolia/instantsearch.js/pull/3779/files#diff-7fd428b5c85af5b43d66125872408774R70) directly on the `SpeechRecognition` mock to avoid to return multiple values from the function.

Reply:
I'm not sure if [this](https://github.com/algolia/instantsearch.js/pull/3779/commits/aa4f1bbb9540fcd8b86d1577e43d42c6600e09fe) is what you meant by storing the listeners directly on the SpeechRecognition mock.